### ISSUE TITLE
String#concat and Array#concat don't support multiple arguments

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1519]
This ability is available since Ruby 2.4

To reproduce:

```ruby
greeting = 'Hello'
greeting.concat(', ', 'World', '!')

array = [1, 2, 3]
array.concat([4, 5], [6], [7, 8, 9])
```

Error:

```
ArgumentError: wrong number of arguments (given 3, expected 1)
	from (irb):28:in `concat'
	from (irb):28
	from ~/.rubies/truffleruby-1.0.0-rc10/bin/irb:29:in `<main>'

ArgumentError: wrong number of arguments (given 3, expected 1)
	from (irb):30:in `concat'
	from (irb):30
	from ~/.rubies/truffleruby-1.0.0-rc10/bin/irb:29:in `<main>'
```

**Comment by chrisseaton** [comment]
Can't see any tagged specs for this - needs specs first.

**Comment by norswap** [comment]
fixed by 6ec3567b4ee3e648858dd72cbedb5e6c69c5625c
